blog: Add "Release Management Post Godot 4.0" (#621)
godotengine/godot-website: collections/_article/release-management-4-1.md

---
title: "Release Management Post Godot 4.0"

> excerpt: "We plan on releasing Godot 4.1 at the end of June after three months of feature merging and one month of bug fixing."

Reviewer: ```suggestion
excerpt: "We outline our plan for future Godot releases and Godot 4.1 specifically, which you can expect at the end of June, after three months of feature merging and one month of bug fixing."
```

categories: ["news"]
author: Clay John
image: /storage/app/uploads/public/638/55e/dad/63855edad2371754303509.png
date: 2023-04-19 14:00:00
---

> Now that [Godot 4.0 is out](https://godotengine.org/article/godot-4-0-sets-sail/) and we have all returned from [GDC](https://godotengine.org/article/gdc-2023-retrospective/) we can finally take the time to solidify our release management plan for the 4.x releases (starting with 4.1). In our [earlier blog post](https://godotengine.org/article/release-management-4-0-and-beyond/) we mentioned that we would like to shift to smaller, more regular releases. Our hope is that we can avoid some of the crunch that comes with trying to squeeze big features into the next release and the resulting long alpha/beta periods that result from cramming in too many big features into a release. 

Reviewer: ```suggestion
Now that [Godot 4.0 is out](https://godotengine.org/article/godot-4-0-sets-sail/) and we have all returned from [GDC](https://godotengine.org/article/gdc-2023-retrospective/) we can finally take the time to solidify our release management plan for the 4.x releases (starting with 4.1). In our [earlier blog post](https://godotengine.org/article/release-management-4-0-and-beyond/) on the matter we mentioned that we would like to shift to smaller, more regular releases.

Our hope is that we can avoid some of the crunch that comes with trying to squeeze bigger features into the upcoming release, not knowing when the next opportunity may appear. This should also help us avoid prolonged beta testing periods, required when too many features are crammed into a single release and even more being constantly added during testing.
```

> Now that [Godot 4.0 is out](https://godotengine.org/article/godot-4-0-sets-sail/) and we have all returned from [GDC](https://godotengine.org/article/gdc-2023-retrospective/) we can finally take the time to solidify our release management plan for the 4.x releases (starting with 4.1). In our [earlier blog post](https://godotengine.org/article/release-management-4-0-and-beyond/) we mentioned that we would like to shift to smaller, more regular releases. Our hope is that we can avoid some of the crunch that comes with trying to squeeze big features into the next release and the resulting long alpha/beta periods that result from cramming in too many big features into a release. 

Reviewer 2: ```suggestion
Now that [Godot 4.0 is out](https://godotengine.org/article/godot-4-0-sets-sail/) and we have all returned from the [GDC](https://godotengine.org/article/gdc-2023-retrospective/) we can finally take the time to solidify our release management plan for the 4.x releases (starting with 4.1). In our [earlier blog post](https://godotengine.org/article/release-management-4-0-and-beyond/) we mentioned that we would like to shift to smaller, more regular releases. Our hope is that we can avoid some of the crunch that comes with trying to squeeze big features into the next release and the resulting long alpha/beta periods that result from cramming in too many big features into a release. 
```

Author: When using the acronym, it is stylized without the "the". So "The Game Developers Conference" is correct, but not "The GDC". See how they stylize it on their about page https://gdconf.com/about

> The current plan is to organize releases into two phases: the feature merging phase and the bug fixing phase. 

Reviewer: ```suggestion
The current plan is to dedicate a set short timeframe to each release, where the development of the next version starts right after the previous one is released. To dedicate time more effectively and provide a focused environment for contributors we want to organize releases into two phases: the feature merging phase and the bug fixing phase.
```

I feel like this clarification is missing here, because below where we talk about the next release being only a few months away, it's not clear IMO why it's a few months away. So this should provide the necessary exposition.

> The current plan is to organize releases into two phases: the feature merging phase and the bug fixing phase. 

Reviewer 2: Next sentence starts with *the current plan* too in a different context. I'm too tired to come up with an alternative, and it doesn't matter much, just wanted to note it :P

Reviewer: I've expanded upon it in my suggestions above. Should address your concern, perhaps? :)

Author: I'll rephrase one of them!

## Godot 4.2

If the above goes well and we release Godot 4.1 by the end of June we will target the end of October for releasing 4.2. Naturally, if the release of 4.1 is delayed, our estimate for 4.2 will be delayed as well as we will still ensure we maintain our 3 month feature merging and 1 month bug fixing phases.

Since this is a new process, don't be surprised if we make adjustments to the plan as we go along. We are still trying to figure out what sort of process works best with our contributors and our users. On that note, we appreciate feedback so please make sure to share your pain points so we can make this release process work for everyone. 

## Long Term Support of Stable Releases 

If we follow the above schedule, we will be making around 3 minor releases per year. We expect that it will quickly become difficult to cherry-pick safe bug fixes back to all previous minor releases. In the Godot 3.x cycle we restricted cherry-picking to only the latest minor release and we only updated older releases when needed to fix critical issues.

> Our intention for the 4.x release cycle is that the upgrade path between releases will be even smoother than in 3.x with less breaking changes, so there will be less reason to remain on older minor releases. Accordingly, for the time being we will be start by restricting cherry-picks to the latest minor release. In practical terms this means that once 4.1 releases our focus will turn to 4.1.1 and 4.2.0 and we won't plan any more 4.0.x releases.

Reviewer: ```suggestion
Our intention for the 4.x release cycle is that the upgrade path between releases will be even smoother than in 3.x with less breaking changes, so there will be less reason to remain on older minor releases. Accordingly, for the time being we will start by restricting cherry-picks to the latest minor release. In practical terms this means that once 4.1 releases our focus will turn to 4.1.1 and 4.2 and we won't plan any more 4.0.x releases.
```

Also, not sure about making promises of a smooth transition. This matter is currently up in the air in terms of how to approach this, so... I'd avoid it if we can.

And I'm not sure we agreed on dropping support for 4.n-1 the moment 4.n is released. This is rather harsh. I think we can support at least 2 minor releases as we develop the next minor release.

Author: I'll rephrase this paragraph to soften it

## Godot 3.6

> We released released the first [beta for Godot 3.6](https://godotengine.org/article/dev-snapshot-godot-3-6-beta-1/) and we plan on releasing Godot 3.6 in the next few months. We don't have an exact release date in mind, but we will be continuing to merge pull requests to the 3.x branch and will release 3.6 when it is ready. 

Reviewer: ```suggestion
We have just released the first [beta for Godot 3.6](https://godotengine.org/article/dev-snapshot-godot-3-6-beta-1/) and we plan on releasing the stable version in the next few months. As the main focus of contributors is on Godot 4 and beyond, the development process for Godot 3 is becoming slower. As such, we don't have an exact release date in mind, but we will be continuing to merge pull requests into the 3.x branch and will release 3.6 when it is ready. 
```

---
title: "Release Management Post Godot 4.0"
excerpt: "We outline our plan for future Godot releases and Godot 4.1 specifically, which you can expect at the end of June, after three months of feature merging and one month of bug fixing."
categories: ["news"]
author: Clay John
image: /storage/blog/covers/release-management-4-1.jpg

> date: 2023-04-19 14:00:00

Reviewer: ```suggestion
date: 2023-04-20 11:00:00
```

---

> Now that [Godot 4.0 is out](https://godotengine.org/article/godot-4-0-sets-sail/) and we have all returned from [GDC](https://godotengine.org/article/gdc-2023-retrospective/) we can finally take the time to solidify our release management plan for the 4.x releases (starting with 4.1). In our [earlier blog post](https://godotengine.org/article/release-management-4-0-and-beyond/) on the matter, we mentioned that we would like to shift to smaller, more regular releases.

Reviewer: ```suggestion
Now that [Godot 4.0 is out](/article/godot-4-0-sets-sail/) and we have all returned from [GDC](/article/gdc-2023-retrospective/) we can finally take the time to solidify our release management plan for the 4.x releases (starting with 4.1). In our [earlier blog post](/article/release-management-4-0-and-beyond/) on the matter, we mentioned that we would like to shift to smaller, more regular releases.
```

> During the 3.x development cycle we noticed two concerning trends:

Reviewer: ```suggestion
During the 4.0 development cycle we noticed two concerning trends:
```

## Godot 3.6

We have just released the first [beta for Godot 3.6](https://godotengine.org/article/dev-snapshot-godot-3-6-beta-1/) and we plan on releasing the stable version in the next few months. As the main focus of contributors is on Godot 4 and beyond, the development process for Godot 3 is becoming slower. As such, we don't have an exact release date in mind, but we will be continuing to merge pull requests into the 3.x branch and will release 3.6 when it is ready. 

Once we release 3.6, we don't intend to make another minor release in the 3.x series. We will continue releasing 3.6.x bug fix updates for as long as we receive contributions to 3.6.

## Godot 5.0?

We have not started planning for Godot 5.0 yet. Our hope is to continue developing Godot 4.x for many years. Instead of planning for breaking changes we will work diligently to add new features without breaking compatibility. We believe the foundation we laid with 4.0 will make this much more tolerable than it was previously.

## Support

> Godot is a non-profit, open source game engine developed by hundreds of contributors on their free time, and a handful of part or full-time developers, hired thanks to [donations from the Godot community](https://godotengine.org/donate). A big thankyou to everyone who has contributed [their time](https://github.com/godotengine/godot/blob/master/AUTHORS.md) or [financial support](https://github.com/godotengine/godot/blob/master/DONORS.md) to the project!
> 
> If you'd like to support the project financially and help us secure our future hires, you can do so on [Patreon](https://www.patreon.com/godotengine) or [PayPal](https://godotengine.org/donate).

Reviewer: ```suggestion
Godot is a non-profit, open source game engine developed by hundreds of contributors on their free time, and a handful of part or full-time developers hired thanks to [donations from the Godot community](/donate). A big thank you to everyone who has contributed [their time](https://github.com/godotengine/godot/blob/master/AUTHORS.md) or [financial support](https://github.com/godotengine/godot/blob/master/DONORS.md) to the project!

If you'd like to support the project financially and help us secure our future hires, you can do so on [Patreon](https://www.patreon.com/godotengine) or [PayPal](/donate).
```
